x/delayedack/keeper/keeper.go

```diff
@@ -70,9 +69,23 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
+func (k Keeper) ExtractChainIDFromChannel(ctx sdk.Context, portID string, channelID string) (string, error) {
+	_, clientState, err := k.channelKeeper.GetChannelClientState(ctx, portID, channelID)
+	if err != nil {
+		return "", fmt.Errorf("failed to extract clientID from channel: %w", err)
+	}
+
+	tmClientState, ok := clientState.(*ibctypes.ClientState)
+	if !ok {
+		return "", nil
+	}
+
+	return tmClientState.ChainId, nil
+}
+
 // GetRollappIDFromPacket retrieves the Rollapp ID from a given packet.
 func (k Keeper) GetRollappIDFromPacket(ctx sdk.Context, packet channeltypes.Packet) (string, error) {
-	rollappID, err := k.rollappkeeper.ExtractRollappIDFromChannel(ctx, packet.DestinationPort, packet.DestinationChannel)
+	rollappID, err := k.ExtractChainIDFromChannel(ctx, packet.DestinationPort, packet.DestinationChannel)
```

Comment:
don't we miss here
```
rollapp, found := im.rollappkeeper.GetRollapp(ctx, chainID)
if !found {
  logger.Debug("Skipping denommetadata middleware. Chain is not a rollapp. ", "chain_id", chainID, "err", err)
  return im.app.OnRecvPacket(ctx, packet, relayer)
}
```
otherwise how it knows whether it's rollapp or not?

Reply:
so it actually the check is after in the middleware code (line 136 in ibc_middleware) but I agree that the name is confusing. It shouldn't be `GetRollappIDFromPacket` but `GetChainIDFromPacket`. Actually this function is redundant alltogether. All it does is calling `ExtractChainIDFromChannel`.  Will fix that.